pyproject.toml

```diff
@@ -31,3 +35,9 @@ plugins = ["pydantic.mypy"]
 [build-system]
 requires = ["poetry-core"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.poetry.extras]
+# used for local tokenization
+tokenizers = ["tokenizers", "requests"]
+# used for caching
+cachetools = ["cachetools"]
```

Comment:
what do these do is it config?

Reply:
yeah just like the concept of dependencies, they are not installed by default when you do `poetry install`